File: JAVA/lecture5/GO/src/GraphicalObject.java
public abstract class GraphicalObject
{

    public GraphicalObject(int color)
    {
        setColor(color);
    }

    public GraphicalObject(GraphicalObject go)
    {
        setColor(go.getColor());
    }

    public GraphicalObject()
    {
        this(0);
    }

    public final int getColor() {
        return color;
    }

    public final void setColor(int color) {
        this.color = color;
    }

    @Override
    public String toString()
    {
        return "[" + getColor() + "]";
    }

    public abstract GraphicalObject kopie();

    public void draw()
    {
//        System.out.println(this);
        System.out.print("[" + getColor() + "]");
    }

    private int color;
}

File: CIS/lectures/lecture4/paty/GraphicalObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ctvrty
{
  internal abstract class GraphicalObject
  {
    public int Color
    {
      get; set;
    }

    public GraphicalObject(int color) => Color = color;

    public abstract void Draw();

    public abstract GraphicalObject CopyOfObject();

  }
}
